sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=protected-access
+
+from typing import (  # pylint: disable=unused-import
+    Any,
+    IO,
+    TYPE_CHECKING,
+)
+import six
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.polling import LROPoller
+from azure.core.polling.base_polling import LROBasePolling  # pylint: disable=no-name-in-module,import-error
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy
+from ._generated._form_recognizer_client import FormRecognizerClient as FormRecognizer
+from ._response_handlers import (
+    prepare_receipt_result,
+    prepare_layout_result,
+)
+from ._generated.models import AnalyzeOperationResult
+from ._helpers import get_content_type, POLLING_INTERVAL, COGNITIVE_KEY_HEADER
+if TYPE_CHECKING:
+    from azure.core.credentials import AzureKeyCredential
+
+
+class FormRecognizerClient(object):
+    """FormRecognizerClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of AzureKeyCredential if using an API key.
+    :type credential: ~azure.core.credentials.AzureKeyCredential
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):
+        # type: (str, AzureKeyCredential, Any) -> None
+        self._client = FormRecognizer(
+            endpoint=endpoint,
+            credential=credential,
+            authentication_policy=AzureKeyCredentialPolicy(credential, COGNITIVE_KEY_HEADER),
+            **kwargs
+        )
+
+    @distributed_trace
+    def begin_extract_receipts(self, stream, **kwargs):
+        # type: (IO[bytes], Any) -> LROPoller
+        """Extract field text and semantic values from a given receipt document.
+        The input document must be of one of the supported content types - 'application/pdf',
+        'image/jpeg', 'image/png' or 'image/tiff'.
+
+        :param stream: .pdf, .jpg, .png or .tiff type file stream.
+        :type stream: stream
+        :keyword bool include_text_details: Include text lines and element references in the result.
+        :keyword str content_type: Media type of the body sent to the API.
+        :return: LROPoller
+        :rtype: ~azure.core.polling.LROPoller
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        if isinstance(stream, six.string_types):
+            raise TypeError("Call begin_extract_receipts_from_url() to analyze a receipt from a url.")
+
+        include_text_details = kwargs.pop("include_text_details", False)
+        content_type = kwargs.pop("content_type", None)
+        if content_type is None:
+            content_type = get_content_type(stream)
+
+        def callback(raw_response, _, headers):  # pylint: disable=unused-argument
```

Comment:
It seems these callbacks are all the same except for the `prepare_receipt_result` etc. Can you consolidate this by creating a function that you just pass those functions to

Reply:
will look into refactoring for the each url/stream method